Tests/ClientTests/Ecosia/EcosiaInstallTypeTests.swift

```diff
@@ -65,6 +67,53 @@ final class EcosiaInstallTypeTests: XCTestCase {
     }
 }
 
+extension EcosiaInstallTypeTests {
+    
+    // Test evaluating install type and version for a fresh install with firstTime=true
+    func testEvaluateFreshInstallType_WithFirstTime_And_VersionProvider() {
+        User.shared.firstTime = true
+        let versionProvider = MockAppVersionInfoProvider(mockedAppVersion: "1.0.0")
+        EcosiaInstallType.evaluateCurrentEcosiaInstallTypeWithVersionProvider(versionProvider)
+        
+        XCTAssertEqual(EcosiaInstallType.get(), .fresh)
+        XCTAssertEqual(EcosiaInstallType.persistedCurrentVersion(), "1.0.0")
+    }
+
+    // Test evaluating install type and version for a fresh install with firstTime=false
+    func testEvaluateFreshInstallType_WithFirstTimeFalse_And_VersionProvider() {
```

Comment:
I guess this is the use case which you introduced `User.shared.firstTime` to fix, right? Because `EcosiaInstallType.persistedCurrentVersion()` is nil since this is the first time the app is opened after `EcosiaInstallType` was implemented, and even so it is not fresh. 

Just double checking since it is a bit counter intuitive 😅 Also, I guess the name is not right since it mentions fresh install?